### ISSUE TITLE
KeyError: 'maxChargerCurrent'

### ISSUE THREAD

**Issue by stigvi** [fondberg/easee_hass#156]
I am running the following service call and get the error below. Does someone know why? This has been working. Have Easee changed something?

service: easee.set_charger_max_limit
data:
  charger_id: EH123456
  current: 6
  
2022-01-15 13:12:14 ERROR (MainThread) [homeassistant.components.websocket_api.http.connection] [2908465112] Error handling message: Unknown error
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/decorators.py", line 27, in _handle_async_response
    await func(hass, connection, msg)
  File "/usr/src/homeassistant/homeassistant/components/websocket_api/commands.py", line 527, in handle_execute_script
    await script_obj.async_run(msg.get("variables"), context=context)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 1260, in async_run
    await asyncio.shield(run.async_run())
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 363, in async_run
    await self._async_step(log_exceptions=False)
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 381, in _async_step
    await getattr(self, handler)()
  File "/usr/src/homeassistant/homeassistant/helpers/script.py", line 584, in _async_call_service_step
    await service_task
  File "/usr/src/homeassistant/homeassistant/core.py", line 1495, in async_call
    task.result()
  File "/usr/src/homeassistant/homeassistant/core.py", line 1530, in _execute_service
    await handler.job.target(service_call)
  File "/config/custom_components/easee/services.py", line 374, in charger_execute_set_current
    charger = controller.check_charger_current(
  File "/config/custom_components/easee/controller.py", line 485, in check_charger_current
    charger_data.state[compareP1] != currentP1
  File "/usr/local/lib/python3.9/site-packages/pyeasee/utils.py", line 54, in __getitem__
    if type(self._storage[key]) == str and validate_iso8601(self._storage[key]):
KeyError: 'maxChargerCurrent'

**Comment by olalid** [comment]
I think this is a bug in the recent versions of the easee integration.
We introduced a sanity check when changing the current limits to avoid calling the cloud API more than necessary.
This specific one seems to have slipped through without proper testing...

But, since you are using this service a word of warning: easee.set_charger_max_limit is not really supposed to be changed often. It will write the configuration to flash internally in the charger and it may wear out if changed often.
If you want to do load balancing and change current with an automation you should use easee.set_circuit_dynamic_limit.
It works the same way but does not write to flash.

**Comment by olalid** [comment]
PS. Easee has some information about how load balancing works here:
https://developer.easee.cloud/docs/load-balancing

**Comment by stigvi** [comment]
> I think this is a bug in the recent versions of the easee integration. We introduced a sanity check when changing the current limits to avoid calling the cloud API more than necessary. This specific one seems to have slipped through without proper testing...
> 
> But, since you are using this service a word of warning: easee.set_charger_max_limit is not really supposed to be changed often. It will write the configuration to flash internally in the charger and it may wear out if changed often. If you want to do load balancing and change current with an automation you should use easee.set_circuit_dynamic_limit. It works the same way but does not write to flash.

I use the dynamic limit service calls to do load balancing. The problem with them is that the current is reset to max when the car is plugged in. So I also set max limit. This service call is used seldom and probably not more than 4-6 times a year.

**Comment by olalid** [comment]
That is not my experience, are you really using the circuit dynamic limit (i.e. not the charger dynamic limit)?

Anyway, I created a PR that I think will fix the problem. Feel free to test if you want, I will probably make a new release later today if all goes well.

**Comment by stigvi** [comment]
> That is not my experience, are you really using the circuit dynamic limit (i.e. not the charger dynamic limit)?
> 
> Anyway, I created a PR that I think will fix the problem. Feel free to test if you want, I will probably make a new release later today if all goes well.

I am using the dynamic circuit  limit to do the load balancing. I have two chargers on the same circuit so I also have to limit the chargers individually so that one charger does not use all available current. It is the charger dynamic limit which is reset when a car is disconnected. But each charger is limited to 6A most of the time and it is rarely changed to 16A if I need fast charging.

**Comment by stigvi** [comment]
I tried to use the dynamic charger limit and made some automation to reduce the current when a car was detected. But my car was always fooled by this and it could not calculate what time it should start charging to be finished at the right time in the morning. So I ended up with setting the max charger current. My car gets this information when plugged in and it is able to calculate charging time based on available current.
